### PR TITLE
Misc CP2K package improvements

### PR DESCRIPTION
**Pull request by dev-zero**
@alalazo ok for you?
@ltalirz FYI

### CODE REVIEW

**Comment by ltalirz** [comment]
@dev-zero thanks, much appreciated!

**Comment by ltalirz** [comment]
By the way, why call this version `develop` when it's the `master` branch you are getting?
If you don't want to call it `master`, there's also `trunk` which is still used in a few places [1](https://www.cp2k.org/version_history) [2](https://manual.cp2k.org/trunk/index.html)

**Comment by dev-zero** [comment]
@ltalirz because it is - speaking in terms of versions - the development version of CP2K, no matter how the branch it originates from is called. `trunk` is the old description of the main branch in SVN. Calling it `master` is also kind of wrong since that is a label for the first branch in a Git repo.

**Comment by dev-zero** [comment]
those test failures look like spurious/unrelated errors

File: var/spack/repos/builtin/packages/cp2k/package.py [hunk 1]
```diff
@@ -216,7 +217,8 @@ def edit(self, spec, prefix):
         if '+mpi' in self.spec:
             cppflags.extend([
                 '-D__parallel',
-                '-D__SCALAPACK'
+                '-D__SCALAPACK',
+                '-D__SCALAPACK2',
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/cp2k/package.py]: I wish people stop using reserved names for their macros :sob:

Comment (dev-zero) [reply var/spack/repos/builtin/packages/cp2k/package.py]: I call that `hysterically grown`

**Comment by dev-zero** [comment]
There will be another PR as soon as this one is merged: https://github.com/dev-zero/spack/compare/feature/cp2k-work...dev-zero:feature/cp2k-sirius

**Comment by ltalirz** [comment]
@dev-zero do you want to open the sirius PR?

**Pull request merged**